force-app/main/interoperability/classes/ProductRegistryMapper_TEST.cls

```diff
@@ -0,0 +1,90 @@
+ */
+@isTest
+private class ProductRegistryMapper_TEST {
+    /**************************************************************************************************************************
+     * @description Test method to verify the getProductRegistrySettingsProductInformation method returns the list
+     * of ProductRegistryModel correctly
+     ***************************************************************************************************************************/
+    @isTest
+    private static void testGetProductRegistrySettingsProductInformation() {
+        String action = 'testAction';
+        String namespace = 'testNamespace';
+        String classname = 'testClassname';
+        Decimal apiVersion = 50.0;
+
+        ProductRegistryMapper.instance = new ProductRegistryMapperStub();
+        List<ProductRegistryModel> prodRegistryModelList;
+
+        Test.startTest();
+        prodRegistryModelList = ProductRegistryMapper.getInstance().getProductRegistrySettingsProductInformation();
+        Test.stopTest();
+
+        System.assert(prodRegistryModelList != null, 'prodRegistryModelList should not be null');
+        System.assert(!prodRegistryModelList.isEmpty(), 'prodRegistryModelList should not be empty');
+        System.assertEquals(1, prodRegistryModelList.size(), 'prodRegistryModelList should have 1 element');
+
+        ProductRegistryModel model = prodRegistryModelList.get(0);
+
+        System.assertEquals(action, model.action, 'action should be set with the value: ' + action);
+        System.assertEquals(namespace, model.namespace, 'namespace should be set with the value: ' + namespace);
+        System.assertEquals(classname, model.classname, 'classname should be set with the value: ' + classname);
+        System.assertEquals(apiVersion, model.apiVersion, 'apiVersion should be set with the value: ' + apiVersion);
+    }
+
+    /**************************************************************************************************************************
+     * @description Test method to comply test coverage because of with the lines not touched because of our mocked classes
+     ***************************************************************************************************************************/
+    @isTest
+    private static void testJustForCoverage() {
+        List<ProductRegistryModel> prodRegistryModelList;
+        Test.startTest();
+        prodRegistryModelList = ProductRegistryMapper.getInstance().getProductRegistrySettingsProductInformation();
+        Test.stopTest();
+    }
+
+    /**************************************************************************************************************************
+     * @description Stub class for ProductRegistryMapper returning some sample data
+     ***************************************************************************************************************************/
+    private class ProductRegistryMapperStub extends ProductRegistryMapper {
```

Comment:
Considering the use case here, this is a valid reason to stub the class in itself.

Reply:
Noted. Thanks!